leap_ec/ops.py

```diff
@@ -419,32 +418,45 @@ def _uniform_crossover(ind1, ind2, p_swap):
         ind2.genome[indices_to_swap] = tmp
 
         return ind1, ind2
-
-    while True:
-        parent1 = next(next_individual)
-        parent2 = next(next_individual)
-        # Return the parents unmodified if we're not performing crossover
-        if np.random.uniform() > p_xover:
-            yield parent1
-            yield parent2
-        else:  # Else do crossover
-            child1, child2 = _uniform_crossover(parent1, parent2, p_swap)
-
-            # Invalidate fitness since the genomes have changed
-            child1.fitness = child2.fitness = None
-
-            yield child1
-            yield child2
-
+    
+    # Accumulate children in a deque to be yielded. This lets us keep them between
+    # generations, or between steady state offspring pipeline calls
+    next_children = collections.deque()
```

Comment:
So, the deque is never emptied unless the user explicitly empties it externally?

This will cause problems with long runs/large genomes, won't it, tantamount to a memory leak?

Not sure how I feel about this as default behavior.

Reply:
Only ever 1-2 individuals are in it at a time, and the second only briefly. I am an advocate for having a parameter to override the behavior, though.